buyer.go

```diff
@@ -371,6 +371,18 @@ func (t *ticketPurchaser) purchase(height int64) error {
 		curPrice := nextStakeDiff
 		couldBuy := math.Floor(balSpendable.ToCoin() / nextStakeDiff.ToCoin())
 
+		// Calculate the remaining tickets that could possibly be
+		// mined in the current window. If couldBuy is greater than
+		// possible amount than reduce to that amount
+		ticketsLeftInWindow := (winSize - int64(t.idxDiffPeriod)) *
+			int64(activeNet.MaxFreshStakePerBlock)
+		if couldBuy > float64(ticketsLeftInWindow) {
+			log.Debugf("The total ticket allotment left in this stakediff window is %v. "+
+				"So this wallet's possible tickets that could be bought is %v so it has been "+
```

Comment:
It looks like this comment string is too long? I think the rest of the files are 84 lines max